Mathlib/GroupTheory/Submonoid/Center.lean

```diff
@@ -104,8 +105,18 @@ equivalence in general; one case when it is is for groups with zero, which is co
 `centerUnitsEquivUnitsCenter`. -/
 @[to_additive (attr := simps! apply_coe_val)
   "For an additive monoid, the units of the center inject into the center of the units."]
-def unitsCenterToCenterUnits [Monoid M] : (Submonoid.center M)ˣ →* Submonoid.center (Mˣ) :=
-  (Units.map (Submonoid.center M).subtype).codRestrict _ <| fun u r ↦ Units.ext <| u.1.prop r
```

Comment:
Is this construction salvageable?

Reply:
@eric-wieser I've got it to something that's much closer to the original now. I can't see a way to improve it further at the moment.